client/app/components/queries/ScheduleDialog.jsx

```diff
@@ -19,6 +19,42 @@ const DATE_FORMAT = 'YYYY-MM-DD';
 const HOUR_FORMAT = 'HH:mm';
 const { Option, OptGroup } = Select;
 
+function TimeEditor(props) {
+  const [time, setTime] = useState(props.defaultValue);
```

Comment:
Why store state and not show the time passed in props?

Reply:
Getting the updated value from prop is also a wrong approach imo, since the value not set externally but from the component itself. Also, value from prop assumes that a certain flow happens outside of the component, which introduces risk of breakage.